src/firefly/js/charts/ChartDataTypeXYCols.js

```diff
@@ -80,119 +89,112 @@ const RESET_ZOOM = `${XYPLOT_DATA_KEY}/RESET_ZOOM`;
  * @prop {string}  [y.options] - comma separated Y axis options: grid, log, and flip
  */
 
+export function getDefaultXYPlotOptions(tbl_id) {
+
+    const {tableMeta, tableData, totalRows}= getTblById(tbl_id);
+
+    if (!totalRows) {
+        return;
+    }
+
+    // for catalogs use lon and lat columns
+    let isCatalog = Boolean(tableMeta[MetaConst.CATALOG_OVERLAY_TYPE] && tableMeta[MetaConst.CATALOG_COORD_COLS]);
+    let xCol = undefined, yCol = undefined;
+
+    if (isCatalog) {
+        const s = tableMeta[MetaConst.CATALOG_COORD_COLS].split(';');
+        if (s.length !== 3) return;
+        xCol = colWithName(tableData.columns, s[0]); // longtitude
+        yCol = colWithName(tableData.columns, s[1]); // latitude
+
+        if (!xCol || !yCol) {
+            isCatalog = false;
+        }
+    }
+
+    // otherwise use the first one-two numeric columns
+    if (!isCatalog) {
+        const numericCols = getNumericCols(tableData.columns);
+        if (numericCols.length >= 2) {
+            xCol = numericCols[0];
+            yCol = numericCols[1];
+        } else if (numericCols.length > 1) {
+            xCol = numericCols[0];
+            yCol = numericCols[0];
+        }
+    }
+
+    return (xCol && yCol) ?
+            {
+                x: {columnOrExpr: xCol.name, options: isCatalog ? 'flip' : '_none_'},
+                y: {columnOrExpr: yCol.name}
+    } : undefined;
+}
+
 /**
- * Load xy plot data Load xy plot data.
+ * Load xy plot data Load xy plot data - left for backward compatibility
  *
  * @param {Object} params - dispatch parameters
  * @param {string} params.chartId - if no chart id is specified table id is used as chart id
  * @param {XYPlotParams} params.xyPlotParams - XY plot options (column names, etc.)
- * @param {boolean} [params.markAsDefault=false] - are the options considered to be "the default" to reset to
  * @param {string} params.tblId - table id
  * @param {Function} [params.dispatcher=flux.process] - only for special dispatching uses such as remote
- * @public
- * @function dispatchLoadPlotData
- * @memberof firefly.action
  */
-export function dispatchLoadPlotData({chartId, xyPlotParams, markAsDefault=false, tblId, dispatcher=flux.process}) {
-    dispatcher({type: LOAD_PLOT_DATA, payload: {chartId: (chartId||tblId), xyPlotParams, markAsDefault, tblId}});
+export function loadXYPlot({chartId, xyPlotParams, tblId, dispatcher=flux.process}) {
+    //SCATTER
+    dispatchChartAdd({chartId, chartType: 'scatter', groupId: tblId,
+        chartDataElements: [
+            {
+                type: 'xycols', //DATATYPE_XYCOLS.id,
+                options: xyPlotParams,
+                tblId
+            }
+        ], dispatcher});
```

Comment:
I am concerned about this as a utility routine. I am not sure we want to make it an example.  It is easier to understand the code if dispatches are being used directly. `loadXYPlot` is effectively a dispatch but you can't tell by looking at the code.  
a couple of other things:
 - `'scatter'` should type a a constant.
 - why not just say `type: DATATYPE_XYCOLS.id`?


Being picky because all this code will be examples for all the other charts.